### PR TITLE
Add `chainerx.log1p`

### PR DESCRIPTION
**Pull request by sky58**
Add `log1p` routine

### CODE REVIEW

**Comment by niboshi** [comment]
Thank you very much!

Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
It seems the test error is the same as this one: https://github.com/chainer/chainer/pull/7126#discussion_r284974443

Could you fix it in a similar manner?

**Comment by niboshi** [comment]
Could you fix test error?

Also I guess this fix (https://github.com/chainer/chainer/pull/7161/commits/b7a3d6a54cd666cb2c64ba604437f0ea129053da) will not work: it's the `std::log(float)` overload that should be avoided.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Please implement `chainerx::cuda::Log1p` in this manner: https://github.com/chainer/chainer/pull/7126/commits/be23ef977b3694daf193e7302e5f0094c58873e4#diff-fc57f97e4c49fc3aa1e97d8209904a20

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

File: chainerx_cc/chainerx/cuda/float16.cuh [hunk 1]
```diff
@@ -71,6 +71,7 @@ public:
     __device__ Float16 Exp() const { return Float16{std::exp(static_cast<float>(*this))}; }
     __device__ Float16 Log() const { return Float16{std::log(static_cast<float>(*this))}; }
     __device__ Float16 Log10() const { return Float16{std::log10(static_cast<float>(*this))}; }
+    __device__ Float16 Log1p() const { return Float16{std::log1p(static_cast<float>(*this))}; }
```
Comment (niboshi) [inline chainerx_cc/chainerx/cuda/float16.cuh]: Can you use `log1pf` here as well?

Comment (sky58) [reply chainerx_cc/chainerx/cuda/float16.cuh]: I resolve this

**Comment by sky58** [comment]
I use `log1pf` in float16.cuh

**Comment by niboshi** [comment]
Thanks!
Jenkins, test this please

**Comment by niboshi** [comment]
Thanks!

**Pull request merged**